File: uHunt/C++src/extra/FallingAnts.cpp
using namespace std;
#include<cstdio>
#include<algorithm>

bool test=true;

int main()
{
    if(test) freopen("../../data/extra/FallingAnts.in","r",stdin);

    int T, L, W, H, rta;
    double max;

    for(scanf("%d",&T); T!=0; scanf("%d",&T))
    {
        int volums[T];
        double accs[T];
        rta=0;max=0.0;

        for(int i=0;i<T;i++)
        {
            scanf("%d %d %d",&L,&W,&H);
            volums[i]=L*W*H;
            accs[i]=(10.0-(10.0/(2*H)));
            max=std::max(accs[i],max);
        }

        for(int i=0;i<T;i++) if(accs[i]==max) rta=std::max(rta,volums[i]);

        printf("%d\n",rta);
    }

    return 0;
}

File: uHunt/src/extra/FallingAnts.java
package extra;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class FallingAnts
{
	private static BufferedReader br;
	private static boolean test=true;

	public static void main(String[] args) throws IOException
	{
		if(test) br=new BufferedReader(new FileReader(new File("./data/extra/FallingAnts.in")));
		else br=new BufferedReader(new InputStreamReader(System.in));
		solve();
		br.close();
	}

	private static void solve() throws IOException
	{
		int L, W, H;
		
		for (int T=Integer.parseInt(br.readLine()); T!=0; T=Integer.parseInt(br.readLine()))
		{
			int rta=0;
			double max=0.0;
			int[] volums=new int[T];
			double[] accs=new double[T];
			String[] aux;

			for(int i=0; i<T; i++)
			{
				aux=br.readLine().split(" ");
				L=Integer.parseInt(aux[0]);
				W=Integer.parseInt(aux[1]);
				H=Integer.parseInt(aux[2]);
				
				volums[i]=L*W*H;
				accs[i]=10.0-(10.0/(2*H));
				max=Math.max(max, accs[i]);
			}
			
			for(int i=0; i<T; i++) if(accs[i]==max) rta=Math.max(rta, volums[i]);
			
			System.out.println(rta);
		}
	}

}